### PR TITLE
KAZOO-4822 conference http api

### PR DESCRIPTION
**Pull request by jamhed**

### CODE REVIEW

File: applications/crossbar/src/modules/cb_conferences.erl [hunk 1]
```diff
@@ -222,8 +234,20 @@ create_conference(Context) ->
...
-load_conference(DocId, Context) ->
-    crossbar_doc:load(DocId, Context, ?TYPE_CHECK_OPTION(<<"conference">>)).
+load_conference(ConfId, Context) ->
+    Realm = kz_util:get_account_realm(cb_context:account_id(Context)),
+    Confs = get_conferences(Realm, ConfId),
+    case participants_info(Confs) of
+        {error, Error} -> cb_context:add_system_error(Error, Context);
+        Participants ->
+            ConfCtx = crossbar_doc:load(ConfId, Context, ?TYPE_CHECK_OPTION(<<"conference">>)),
+            ConfInfo = [
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_conferences.erl]: Is this a list of 6 items or is it supposed to be a JSON object with 3 key/value pairs? It seems strange to set `_read_only` with a list of 6 items.

File: applications/crossbar/src/modules/cb_conferences.erl [hunk 2]
```diff
@@ -233,38 +257,47 @@ load_conference(DocId, Context) ->
...
+    kapps_util:amqp_pool_collect(Req, fun kapi_conference:publish_search_req/1, {'ecallmgr', 'true'}).
 
 -spec conference_participants(kz_json:objects()) -> kz_json:objects().
 conference_participants(JObjs) ->
-    [Participant || JObj <- JObjs
-                    ,Participant <- kz_json:get_value(<<"Participants">>, JObj, [])
-    ].
+    lists:flatten([ kz_json:get_value(<<"Participants">>, JObj, []) || JObj <- JObjs ]).
+
+conference_runtime({ok, JObjs}) when is_list(JObjs) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_conferences.erl]: specs needed

File: applications/crossbar/src/modules/cb_conferences.erl [hunk 3]
```diff
@@ -364,7 +399,30 @@ on_successful_validation(DocId, Context) ->
...
+    Conf = kz_json:get_value(<<"value">>, JObj),
+    Stats = conference_stats(JObj, Conf),
+    ExtConf = lists:foldr(fun({K,V}, ExtConf) -> kz_json:set_value(K, V, ExtConf) end, Conf, Stats),
+    [ExtConf | Acc].
+
+-spec conference_stats(kz_json:object(), kz_json:object()) -> kz_json:object().
+conference_stats(JObj, Conf) ->
+    ConfId = kz_json:get_value(<<"id">>, JObj),
+    Realm = kz_util:get_account_realm(kz_json:get_value(<<"owner_id">>, Conf)),
+    {ok, Confs} = get_conferences(Realm, ConfId),
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_conferences.erl]: tickie the atom ;)

File: applications/crossbar/src/modules/cb_conferences.erl (hunk 1 above)
Comment (jamhed) [inline applications/crossbar/src/modules/cb_conferences.erl]: @jamesaimonetti It's supposed to be an object with 3 k/v, and it is (http returns proper object).

File: applications/crossbar/src/modules/cb_conferences.erl [hunk 4]
```diff
@@ -22,11 +22,24 @@
          ,delete/2
         ]).
 
--include("crossbar.hrl").
+% -include("crossbar.hrl").
+-include_lib("crossbar/src/crossbar.hrl").
```
Comment (fenollp) [inline applications/crossbar/src/modules/cb_conferences.erl]: Revert this, it wrecks your build

File: applications/crossbar/src/modules/cb_conferences.erl [hunk 5]
```diff
@@ -22,11 +23,23 @@
          ,delete/2
         ]).
 
--include("crossbar.hrl").
+-include_lib("crossbar/src/crossbar.hrl").
```
Comment (lazedo) [inline applications/crossbar/src/modules/cb_conferences.erl]: why are we changing this ? other cb modules use `include("crossbar.hrl")`

Comment (jamhed) [inline applications/crossbar/src/modules/cb_conferences.erl]: @lazedo: i want to change them all eventually. with -include sync and other similar tools are unable to recompile the module (they don't know where to searh for hrl, but with include_lib they know (code:lib_dir(app)). this works nicely with relx development releases.

Comment (lazedo) [inline applications/crossbar/src/modules/cb_conferences.erl]: then do it in a pr not this one. what you're proposing needs to be debated with the reswt of the team. maybe sync is not the right tool or needs modifications.

Comment (jamhed) [inline applications/crossbar/src/modules/cb_conferences.erl]: @lazedo what should be debated? it breaks nothing and adds possibility.

File: applications/crossbar/src/modules/cb_conferences.erl [hunk 6]
```diff
@@ -215,15 +233,40 @@ create_conference(Context) ->
...
+        {error, Error} -> cb_context:add_system_error(Error, Context);
+        Participants ->
+            ConfCtx = crossbar_doc:load(ConfId, Context, ?TYPE_CHECK_OPTION(<<"conference">>)),
+            ConfInfo = kz_json:from_list([
+                {<<"participants">>, Participants}
+                , {<<"created">>, conference_starttime(Confs)}
+                , {<<"duration">>, conference_runtime(Confs)}
+            ]),
+            cb_context:set_resp_data(ConfCtx, kz_json:set_value(<<"_read_only">>, ConfInfo, cb_context:resp_data(ConfCtx)))
+    end.
```
Comment (lazedo) [inline applications/crossbar/src/modules/cb_conferences.erl]: why this change here ?
